lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -1,38 +1,85 @@
 import { createElement } from 'preact';
 import classNames from 'classnames';
-import propTypes from 'prop-types';
 import { useContext, useEffect, useState, useRef } from 'preact/hooks';
+import propTypes from 'prop-types';
 
 import { Config } from '../config';
-import { submitGrade } from '../utils/grader-service';
-import { formatToNumber, validateGrade } from '../utils/validation';
 import ErrorDialog from './ErrorDialog';
 import Spinner from './Spinner';
 import SvgIcon from './SvgIcon';
+import { fetchGrade, submitGrade } from '../utils/grader-service';
 import { trustMarkup } from '../utils/trusted';
+import { formatToNumber, validateGrade } from '../utils/validation';
 import ValidationMessage from './ValidationMessage';
 
+// Grades are always stored as a value between [0-1] on the service layer.
+// Scale the grade to [0-10] when loading it into the UI and scale it
+// back to [0-1] when saving it to the api service.
+const GRADE_MULTIPLIER = 10;
+
+/**
+ * Custom useEffect function that handles fetching a student's
+ * grade and returning the result of that grade and the loading
+ * and error status of that fetch request.
+ */
+const useFetchGrade = student => {
```

Comment:
I took this opportunity to make a custom hook, I think this works well here because its complicated enough where dumping all this in component makes it harder to read but all we really care about here is the `grade` it gets and is the `gradeLoading` currently?  I also export the `gradeError` but its not used (see below)